src/main/java/uk/gov/companieshouse/service/DissolutionPatcher.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.companieshouse.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.mapper.DirectorApprovalMapper;
+import uk.gov.companieshouse.mapper.DissolutionResponseMapper;
+import uk.gov.companieshouse.model.db.DirectorApproval;
+import uk.gov.companieshouse.model.db.Dissolution;
+import uk.gov.companieshouse.model.db.DissolutionDirector;
+import uk.gov.companieshouse.model.dto.DissolutionPatchResponse;
+import uk.gov.companieshouse.model.enums.ApplicationStatus;
+import uk.gov.companieshouse.repository.DissolutionRepository;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+@Service
+public class DissolutionPatcher {
+
+    private final DissolutionRepository repository;
+    private final DissolutionResponseMapper responseMapper;
+    private final DirectorApprovalMapper approvalMapper;
+
+    @Autowired
+    public DissolutionPatcher(
+        DissolutionRepository repository,
+        DissolutionResponseMapper responseMapper,
+        DirectorApprovalMapper approvalMapper) {
+        this.repository = repository;
+        this.responseMapper = responseMapper;
+        this.approvalMapper = approvalMapper;
+    }
+
+    public DissolutionPatchResponse patch(String companyNumber, String userId, String ip, String email) {
+        final Dissolution dissolution = this.repository.findByCompanyNumber(companyNumber).get();
+        final DissolutionDirector director = this.findDirector(email, dissolution);
+        if (director != null) {
```

Comment:
director should never be null as we should have already validated in the controller that they are a director who is pending approval